cmd/changelog.go

```diff
@@ -80,12 +81,23 @@ var changelogCmd = &cobra.Command{
 	},
 }
 
-// GetCommitLog takes a commit object and returns a commit log, for example:
-// * [2847a2e6](../../commit/2847a2e624ee6736b43cc3a68acd75368d1a75d1) A commit message
+// GetCommitLog takes a commit object and returns a commit log that may link to a pull request, for example:
+// * [2847a2e6](../../commit/2847a2e624ee6736b43cc3a68acd75368d1a75d1) A commit message 
+// * [2847a2e6](../../commit/2847a2e624ee6736b43cc3a68acd75368d1a75d1) A commit message ([#100](../../pull/100))
 func GetCommitLog(commit *object.Commit) string {
 	hash := commit.Hash.String()
 	if hash != "" {
-		return fmt.Sprintf("* [%s](../../commit/%s) %s", hash[:8], hash, strings.Split(commit.Message, "\n")[0])
+		var commitLog string
+		commitMsg := strings.Split(commit.Message, "\n")[0]
+		r := regexp.MustCompile("\\(#\\d+\\)$")
+		idx := r.FindStringIndex(commitMsg)
+		if idx == nil {
+			commitLog = fmt.Sprintf("* [%s](../../commit/%s) %s", hash[:8], hash, commitMsg)
+		} else {
+			prNum := commitMsg[idx[0]+2:idx[1]-1]
```

Comment:
this feels a little weird, why the `+2` and the `-1`?

Reply:
it might make sense to adjust the capture group on the regex so you don't have to do this

Reply:
Because I'm matching the (#\d+)$ pattern to make sure that it matches the PR number and not any other number, the `+2` and `-1` is to extract the PR number within the match.

Reply:
I wasn't sure how else to adjust the capture group as this seemed like the cleanest way to me (otherwise by checking a number I'd have to first check that it corresponds to a pull request number, and then get the last match for all numbers in the commit message)